File: Day20/Challenge3.py
#Challenge - Prime Factors

"""
The program must accept an integer N as the input. The program must print the prime factors of N in
ascending order as the output.
Boundary Condition(s):
10 <= N <= 10^5
Input Format:
The first line contains N.
Output Format:
The first line contains the prime factors of N in ascending order.
Example Input/Output 1:
Input:
100
Output:
2 5
Explanation:
The factors of 100 are 1, 2, 4, 5, 10, 20, 25, 50 and 100.
The prime factors of 100 are 2 and 5.
So 2 and 5 are printed as the output.
Example Input/Output 2:
Input:
150
Output:
2 3 5
"""

#Solution

def isprime(n):
    if n<=1:
        return False
    else:
        for i in range(2, int(n**0.5)+1):
            if(n%i==0):
                return False
        return True

n= int(input())
for i in range(1, n+1):
    if(n%i==0):
        if(isprime(i)):
            print(i, end=" ");

File: Day29/Challenge3.java
//Challenge - String from Digits

/*
The program must accept an integer N and ten alphabets as the input. The program must form a string S based on the following 
condition.
- For each digit D in N, the program must replace the digit D by the alphabet at the index D in the given ten alphabets.
Boundary Condition(s):
1 <= N <= 10^8
Input Format:
The first line contains N.
The second line contains ten alphabets separated by a space.
Output Format:
The first line contains the string S as per the given condition.
Example Input/Output 1:
Input:
1870
a b c d e f g h i j
Output:
biha
Explanation:
The 1st digit in 1870 is 1, which is replaced by the alphabet at the index 1 (b).
The 2nd digit in 1870 is 8, which is replaced by the alphabet at the index 8 (i).
The 3rd digit in 1870 is 7, which is replaced by the alphabet at the index 7 (h).
The 4th digit in 1870 is 0, which is replaced by the alphabet at the index 0 (a).
Hence the output is biha
Example Input/Output 2:
Input:
15733
r f g i o m m p q r
Output:
fmpii
*/

//Solution
import java.util.*;
public class Hello {

    public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int n= sc.nextInt();
		sc.nextLine();
		String[] s= sc.nextLine().split(" ");
		System.out.print(replace(n, s));
	}
	
	public static String replace(int n, String s[]){
	    String digits = String.valueOf(n);
	    String res="";
	    for(int i=0; i<digits.length(); i++){
	        int index= Character.getNumericValue(digits.charAt(i));
	        res+=s[index];
	    }
	    return res;
	}
}
